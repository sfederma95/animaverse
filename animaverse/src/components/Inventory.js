import React from 'react';
import Item from './Item';

function Inventory(){
    return(
        <div>
            <h1>Username's inventory</h1>
            <p>Select an item you'd like to use on Ratty.</p>
            <p>Using an item will remove it from your inventory.</p>
            <div>
                <Item id={1} src='something.png' action='Feed' amount={10} name="Candy" description="A little sweet, but who doesn't like candy?" />
                <Item id={2} src='something.png' action='Play' amount={10} name="Ball" description="Time for a little game of fetch!" />
            </div>
        </div>
    )
}

export default Inventory;