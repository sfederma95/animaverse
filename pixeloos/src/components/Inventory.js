import React, {useContext, useEffect} from 'react';
import Item from './Item';
import items from '../items/items'
import UserContext from '../users/UserContext'
import { v4 as uuidv4 } from 'uuid';
import './inventory.css';
import AnimalsApi from '../api'
import jwt from 'jsonwebtoken'

function Inventory(){
    const {currentUser, setCurrentUser} = useContext(UserContext);
    const token = AnimalsApi.token
    useEffect(function loadUserInfo(){
        async function getCurrentUser(){
          if(token) {
            try {
              let {id} = jwt.decode(token);
              AnimalsApi.token = token;
              let getUser = await AnimalsApi.getUser(id);
              setCurrentUser(getUser);
            } catch(err){
              setCurrentUser(null)
            }
          }
        }
        getCurrentUser();
      },[setCurrentUser, token])
    const inventoryItems = currentUser.inventory.map(i=>{
        let currItem = items[i.item_id-1]
        return <Item key={uuidv4()} src={currItem.img} name={currItem.name} description={currItem.description} id={currItem.id} action={currItem.action} amount={currItem.amount} />
    })
    return(
        <div id='inventory'>
            {inventoryItems}
        </div>
    )
}

export default Inventory;