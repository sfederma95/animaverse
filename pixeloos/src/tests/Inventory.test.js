import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import Inventory from '../components/Inventory';
import {UserProvider} from './testSetup';

it ('renders inventory component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <Inventory />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <Inventory/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})