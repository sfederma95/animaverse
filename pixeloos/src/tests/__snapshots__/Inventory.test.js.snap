// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="inventory"
  >
    <div
      class="inv-item"
      id="1"
    >
      <img
        alt="Hamburger"
        class="item-img"
        src="hamburger.png"
        tabindex="0"
      />
    </div>
    <div
      class="inv-item"
      id="8"
    >
      <img
        alt="Hotdog"
        class="item-img"
        src="hotdog.png"
        tabindex="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
